libfuse/start.go

```diff
@@ -42,24 +42,38 @@ func Start(mounter Mounter, options StartOptions, kbCtx libkbfs.Context) *libfs.
 	if err != nil {
 		return libfs.MountError(err.Error())
 	}
-	defer c.Close()
-
-	onInterruptFn := func() {
-		select {
-		case <-c.Ready:
-			// Was mounted, so try to unmount if it was successful.
-			if c.MountError == nil {
-				err = mounter.Unmount()
-				if err != nil {
-					return
+	var doneChan <-chan struct{}
+	var onInterruptFn func()
+
+	if c != nil {
+		defer c.Close()
+
+		doneChan = c.Ready
+		onInterruptFn = func() {
+			select {
+			case <-c.Ready:
+				// Was mounted, so try to unmount if it was successful.
+				if c.MountError == nil {
+					err = mounter.Unmount()
+					if err != nil {
+						return
+					}
 				}
-			}
 
-		default:
-			// Was not mounted successfully yet, so do nothing. Note that the mount
-			// could still happen, but that's a rare enough edge case.
+			default:
+				// Was not mounted successfully yet, so do nothing. Note that the mount
+				// could still happen, but that's a rare enough edge case.
+			}
+			libkbfs.Shutdown()
+		}
+	} else {
+		// When there's no mount, declare ourselves done when we get
+		// an interrupt.
+		ch := make(chan struct{}, 1)
+		doneChan = ch
+		onInterruptFn = func() {
+			ch <- struct{}{}
```

Comment:
maybe send on `ch` only if it's not already full? (To avoid leaving goroutines around if interrupted multiple times)